dirsrvtests/tests/suites/indexes/regression_test.py

```diff
@@ -594,6 +594,181 @@ def test_reindex_extended_matching_rule(topo, add_backend_and_ldif_50K_users):
     ) == 0
 
 
+def test_update_eq_index_after_deleting_and_readding_attribute_in_one_step(topo):
+    """ Test that 'eq' index is properly updated
+    when deleting and re-adding an attribute in one step
+
+    :id: d306b511-01bd-4400-96fa-9838a310f086
+    :setup: Standalone instance
+    :steps:
+        1. Create a user with a 3 valued attribute (mail)
+        2. Modify the user in one step: delete two mail attribute values and re-add one of the values back
+        3. Search for an entry using deleted value as filter
+        4. Search for an entry using the re-added value as filter
+        5. Delete all mail attribute values
+        6. Search for an entry using the re-added value as filter
+        7. Search for an entry using 'mail=*' as a filter
+        8. Re-add two values for mail attribute in order to test with different amount of original values
+        9. Rerun modification in step 2
+        10. Search for an entry using deleted value as filter
+        11. Search for an entry using the re-added value as filter
+    :expectedresults:
+        1. Success
+        2. Success
+        3. No entry found
+        4. Original user found
+        5. Success
+        6. No entry found
+        7. No entry found
+        8. Success
+        9. Success
+        10. No entry found
+        11. Original user found
+    """
+    
+    rdn = 'user0099'
+    name = 'Test User'
+    users = UserAccounts(topo.standalone, DEFAULT_SUFFIX)
+    user = users.create(properties={
+        'uid': rdn,
+        'sn': rdn,
+        'cn': rdn,
+        'uidNumber': '10099',
+        'gidNumber': '10099',
+        'givenname': name,
+        'gecos': name,
+        'description': name,
+        'homeDirectory': '/home/{}'.format(rdn),
+        'mail': ['{}@dev.null'.format(rdn),
+                 'alias@dev.null',
+                 '{}@redhat.com'.format(rdn)]
+        })
+
+    #
+    # Use of modify_s necessary as the test case demands delete and add operation to be done in one operation
+    #
+    try:
+        topo.standalone.modify_s(user.dn, [(ldap.MOD_DELETE,
+                                              'mail',
+                                              b'user0099@dev.null'),
+                                             (ldap.MOD_DELETE,
+                                              'mail',
+                                              b'alias@dev.null'),
+                                             (ldap.MOD_ADD,
+                                              'mail', b'user0099@dev.null')], escapehatch='i am sure')
+    except ldap.LDAPError as e:
+        log.fatal('Failed to modify user: {}'.format(e))
+        assert False
+
+    #
+    # Search using deleted attribute value - no entries should be returned
+    #
+    try:
+        entry = user.search(filter='mail=alias@dev.null')
+        if entry:
```

Comment:
Minor nitpick, but the use is inconsistent across the test.
It's either
- `if entry` and `if not entry`
Or
- `if entry is None` and `if entry is not None`